src/ensRegistry.ts

```diff
@@ -46,16 +49,50 @@ function makeSubnode(event:NewOwnerEvent): string {
   return crypto.keccak256(concat(event.params.node, event.params.label)).toHexString()
 }
 
+function recurseDomainDelete(domain: Domain): string {
+  if (domain.parent !== null) {
+    const parentDomain = Domain.load(domain.parent!)
+    let resolverIsNull = false
+    if (domain.resolver !== null) {
```

Comment:
You can also just return early here if `domain.resolver === null` - that way you don't need to check it in an if statement later.